runtime/parser2/expression.go

```diff
@@ -373,6 +374,42 @@ func parseNumber(token lexer.Token) ast.Expression {
 	}
 }
 
+func defineInvocationExpression() {
+	setExprLeftDenotation(
+		lexer.TokenParenOpen,
+		func(p *parser, token lexer.Token, left ast.Expression) ast.Expression {
+			var arguments []*ast.Argument
+			var endPos ast.Position
+			atEnd := false
+			for !atEnd {
+				p.skipSpaceAndComments(true)
+				switch p.current.Type {
+				case lexer.TokenComma:
```

Comment:
I realized this allows multiple commas and multiple arguments to follow each other, e.g. `foo, ,, bar` or `foo bar`, so this needs a state flag to check if a comma or an argument is expected.

I fixed this for parameter lists here:
https://github.com/onflow/cadence/pull/85/files#diff-74be485a73d45cdb0cc03c3aa0ecfe09R179-R220

Reply:
Good catch! Will implement!